### PR TITLE
refactor(ui): Make Navigating DataHub UI easier, fix duplicate tracking, duplicate networks calls, + misc optimizations

### PR DESCRIPTION
**Pull request by jjoyce0510**
## Summary

In this PR we introduce a hand-full of small changes to improve the navigational experience on DataHub. 

1. Switch `history.push` to `history.replace` in following cases: Changing filters inside of entity page, searching inside schema, switching tabs on entity page, switching tabs on settings view, rebooting initial path on page load, page not found warnings. No more need to spam the back button just to get back to search results page. 
2. [Fix - Perf] Grab the current user from the local context instead of using legacy `useGetAuthenticatedUser` (replaced). This removes a ton of `getMe` queries appearing as you navigate the UI. User is loaded once on boot.  
3. [Fix] Fix double-fetching of Schema History when schema tab loads
4. [Fix] Merge datasets in Apollo Cache to avoid purging the cache on tab switches (Queries -> Schema both cache datasets) 
5. [optimization] remove recommendations from entity profiles - currently they are UNUSED.
6. [Fix] Fix double-calling of expensive searchAcrossLineage query due to a replace in the URL bar to add filters for numHops. 
7. [fix] Remove duplicate tracking events for rendering homepage recommendations and `HomePageViewEvent`

## Status

Ready for review

## Checklist

- [ ] The PR conforms to DataHub's [Contributing Guideline](https://github.com/datahub-project/datahub/blob/master/docs/CONTRIBUTING.md) (particularly [Commit Message Format](https://github.com/datahub-project/datahub/blob/master/docs/CONTRIBUTING.md#commit-message-format))
- [ ] Links to related issues (if applicable)
- [ ] Tests for the changes have been added/updated (if applicable)
- [ ] Docs related to the changes have been added/updated (if applicable). If a new feature has been added a Usage Guide has been added for the same.
- [ ] For any breaking change/potential downtime/deprecation/big changes an entry has been made in [Updating DataHub](https://github.com/datahub-project/datahub/blob/master/docs/how/updating-datahub.md)

### CODE REVIEW

File: datahub-web-react/src/App.tsx [hunk 1]
```diff
@@ -61,7 +61,19 @@ const errorLink = onError((error) => {
...
     link: errorLink.concat(httpLink),
-    cache: new InMemoryCache(),
+    cache: new InMemoryCache({
+        typePolicies: {
+            Query: {
+                fields: {
+                    dataset: {
+                        merge: (oldObj, newObj) => {
+                            return { ...oldObj, ...newObj };
+                        },
```
Comment (chriscollins3456) [inline datahub-web-react/src/App.tsx:71]: and going from one dataset page to another (whether that be through the UI or hitting a new link) won't result in dataset data being merged right?

File: datahub-web-react/src/app/context/useInitialRedirect.ts [hunk 2]
```diff
@@ -19,8 +19,8 @@ export function useInitialRedirect(state, localState, setState, setLocalState) {
                 ...state,
                 loadedInitialPath: true,
             });
-            if (localState.selectedPath) {
-                history.push({
+            if (localState.selectedPath && !localState.selectedPath.includes(PageRoutes.EMBED)) {
```
Comment (chriscollins3456) [inline datahub-web-react/src/app/context/useInitialRedirect.ts:22]: amazing - forgot to bring this to OSS after making this change in saas - thanks!

File: datahub-web-react/src/app/entity/container/ContainerEntity.tsx [hunk 3]
```diff
@@ -99,9 +98,9 @@ export class ContainerEntity implements Entity<Container> {
                 {
                     component: SidebarDomainSection,
                 },
-                {
-                    component: SidebarRecommendationsSection,
-                },
+                // {
+                //    component: SidebarRecommendationsSection,
+                // },
```
Comment (chriscollins3456) [inline datahub-web-react/src/app/entity/container/ContainerEntity.tsx:103]: should we just delete here and other places? or we could leave a comment as to why we we're leaving it

File: datahub-web-react/src/app/entity/shared/tabs/Lineage/generateUseSearchResultsViaRelationshipHook.ts [hunk 4]
```diff
@@ -41,6 +45,7 @@ export default function generateUseSearchResultsViaRelationshipHook({
             variables: {
                 input: inputFields,
             },
+            skip: !filtersExist(filters, orFilters), // If you don't include any filters, we shound't return anything :). Might as well skip!
```
Comment (chriscollins3456) [inline datahub-web-react/src/app/entity/shared/tabs/Lineage/generateUseSearchResultsViaRelationshipHook.ts:48]: only concern is that if we ever remove default filters from this tab (ie. we direct you to have 1 degree or whatever first right now) then we won't fire any query

File: datahub-web-react/src/app/home/HomePage.tsx [hunk 5]
```diff
@@ -13,7 +13,9 @@ import {
 } from '../onboarding/config/HomePageOnboardingConfig';
 
 export const HomePage = () => {
-    analytics.event({ type: EventType.HomePageViewEvent });
+    useEffect(() => {
+        analytics.event({ type: EventType.HomePageViewEvent });
+    }, []);
```
Comment (chriscollins3456) [inline datahub-web-react/src/app/home/HomePage.tsx:18]: yes

**Review by chriscollins3456 — APPROVED** [review]
great improvements!!

File: datahub-web-react/src/App.tsx (hunk 1 above)
Comment (jjoyce0510) [reply datahub-web-react/src/App.tsx:71]: thats right - its per urn!

File: datahub-web-react/src/app/entity/shared/tabs/Lineage/generateUseSearchResultsViaRelationshipHook.ts (hunk 4 above)
Comment (jjoyce0510) [reply datahub-web-react/src/app/entity/shared/tabs/Lineage/generateUseSearchResultsViaRelationshipHook.ts:48]: yep - agree

Comment (jjoyce0510) [reply datahub-web-react/src/app/entity/shared/tabs/Lineage/generateUseSearchResultsViaRelationshipHook.ts:48]: problem right now is worse - we fetch searchAcrosslineage (our most expensive query) 2 times!

File: datahub-web-react/src/app/context/useInitialRedirect.ts (hunk 2 above)
Comment (jjoyce0510) [reply datahub-web-react/src/app/context/useInitialRedirect.ts:22]: that makes sense

**Pull request merged**